Fix a bunch of lint errors

balloob opened the pull request:
This will fix a whole bunch of lint errors.

Related to #1043 

Open issues:
 - How to define the `content` property on the `ha-markdown` custom element
 - How to have the `localize` behavior expose the `localize` function. Looks like the polymer analyzer is unable to see that we also apply the `Polymer.AppLocalizeBehavior` behavior.

balloob committed:
Fix a bunch of lint errors

c727 commented on src/components/entity/ha-chart-base.html:
```diff
@@ -151,6 +151,21 @@
+        },
+        tooltip: {
+          type: Object,
+          value: () => ({
```
why do you use arrow functions here and for metas?

balloob replied:
Objects are mutable data structures. Since we override properties of it, those properties would also be set on future instances of this web component.

If you define a function, it will be invoked to generate a default value, meaning each instance gets their own version of the object.

armills commented:
I think these directives will fix the localize linting problems.

https://github.com/home-assistant/home-assistant-polymer/blob/e0ca88b3add7eafdc36f14c55764adb07a457b99/src/components/ha-sidebar.html#L162-L165

balloob committed:
Add behavior jsdoc

balloob committed:
Convert ha-markdown to polymer element

balloob committed:
Update lint command

balloob commented:
Alright, fixed `ha-markdown` lint errors by converting it to a Polymer element. Added the behavior annotations and updated the lint command. This should make sure we lint correctly moving forward.
